sycl/source/detail/device_info.hpp

```diff
@@ -619,12 +660,30 @@ struct get_device_info_impl<
               ext::oneapi::experimental::info::device::architecture>::value,
           sizeof(DeviceIp), &DeviceIp, nullptr);
       return ReturnHelper(MapDeviceIpToArch, DeviceIp);
+    } else if (Dev->is_gpu() && (backend::ext_oneapi_cuda == CurrentBackend ||
+                                 backend::ext_oneapi_hip == CurrentBackend)) {
+      auto MapArchToDevice = [](const char *arch) {
+        ARCHES(CMP);
+        throw sycl::exception(
+            make_error_code(errc::runtime),
+            "The current device architecture is not supported by "
+            "sycl_ext_oneapi_device_architecture.");
+      };
+      size_t resultSize = 0;
+      Dev->getPlugin()->call<PiApiKind::piDeviceGetInfo>(
+          Dev->getHandleRef(), PiInfoCode<info::device::version>::value, 0,
+          nullptr, &resultSize);
+
+      std::unique_ptr<char[]> DeviceArch(new char[resultSize]);
+      Dev->getPlugin()->call<PiApiKind::piDeviceGetInfo>(
+          Dev->getHandleRef(), PiInfoCode<info::device::version>::value,
+          resultSize, DeviceArch.get(), nullptr);
+      return MapArchToDevice(DeviceArch.get());
     } else if (Dev->is_cpu() && backend::opencl == CurrentBackend) {
       // TODO: add support of different CPU architectures to
       // sycl_ext_oneapi_device_architecture
       return sycl::ext::oneapi::experimental::architecture::x86_64;
     } // else is not needed
-    // TODO: add support of other arhitectures by extending with else if
```

Comment:
Please keep this line, there are other architectures which we can support.

Reply:
Done